Source/SwiftLintFramework/Rules/Performance/ReduceBooleanRule.swift

```diff
@@ -0,0 +1,51 @@
+import SourceKittenFramework
+
+public struct ReduceBooleanRule: Rule, ConfigurationProviderRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "reduce_boolean",
+        name: "Prefer allSatisfy or contains over reduce(true) or reduce(false)",
+        description: "Prefer using `.allSatisfy()` or `.contains()` over `reduce(true)` or `reduce(false)`",
+        kind: .performance,
+        minSwiftVersion: .fourDotTwo,
+        nonTriggeringExamples: [
+            "nums.reduce(0) { $0.0 + $0.1 }",
+            "nums.reduce(0.0) { $0.0 + $0.1 }"
+        ],
+        triggeringExamples: [
+            "let allNines = nums.↓reduce(true) { $0.0 && $0.1 == 9 }",
+            "let anyNines = nums.↓reduce(false) { $0.0 || $0.1 == 9 }",
+            "let allValid = validators.↓reduce(true) { $0 && $1(input) }",
+            "let anyValid = validators.↓reduce(false) { $0 || $1(input) }",
+            "let allNines = nums.↓reduce(true, { $0.0 && $0.1 == 9 })",
+            "let anyNines = nums.↓reduce(false, { $0.0 || $0.1 == 9 })",
+            "let allValid = validators.↓reduce(true, { $0 && $1(input) })",
+            "let anyValid = validators.↓reduce(false, { $0 || $1(input) })"
+        ]
+    )
+
+    public func validate(file: File) -> [StyleViolation] {
+        let pattern = "\\breduce\\((true|false)"
+        return file
+            .match(pattern: pattern,
+                   excludingSyntaxKinds: SyntaxKind.commentAndStringKinds)
```

Comment:
This pattern should always have the same two syntax kinds. We should match for those instead of excluding comments/strings.